src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.Ssl.cs

```diff
@@ -77,7 +79,9 @@ internal static SafeSslHandle SSLStreamCreateWithCertificates(ReadOnlySpan<byte>
             string targetHost)
         {
             int ret = SSLStreamSetTargetHostImpl(sslHandle, targetHost);
-            if (ret != SUCCESS)
+            if (ret == UNSUPPORTED_API_LEVEL)
+                throw new PlatformNotSupportedException("The operation is not supported on this API level");
```

Comment:
Should the message go in resources?

Reply:
Yes it should. Maybe we should also make it more specific:

```suggestion
                throw new PlatformNotSupportedException("Setting an SNI hostname is not supported on this API level.");
```